### PR TITLE
Added JSON data test for randomisers

### LINKED ISSUE #36: Feat: create a JSON object of user data for testing

### LINKED ISSUE #49: create a json schema
to separate calculations and build, I want a json schema of all the features in the randomizer

Something like an array of objects, where each object is similar to this. Missing values can be `null`.
```
{
  description: "",
  "not-available-description": "",
  buttons: [{
    text: '',
    "checklist options": [{
      text: '',
      'not-available-description': ''
    }, {
      text: '',
      'not-available-description': ''
    }]
  }]
}
```

### PR DESCRIPTION
**Pull request by aswinikalyan30**
Closes #36

### CODE REVIEW

**Comment by ieahleen** [comment]
can you add an .md file in there that describe at what api call each file correspond?

**Comment by aswinikalyan30** [comment]
Sure!

**Comment by aswinikalyan30** [comment]
Added the description too!

File: json_files/jsonFiles.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+## Description of JSON files
+### This folder contains the json schema for the responses of the API calls made by habitican-randomiser
+
+* Equip [Random Pet](equipPet.json) : This JSON file contains the schema for equpping a random pet from habitica. 
+
+The API call it responds to is: ```https://habitica.com/api/v3/user/equip/pet/{pet_name}```
```
Comment (ieahleen) [inline json_files/jsonFiles.md:6]:
```suggestion
* Equip [Random Pet](equipPet.json) : This JSON file contains the schema for equpping a random pet from habitica. 

The API call it responds to is: `https://habitica.com/api/v3/user/equip/pet/{pet_name}`
```

File: json_files/jsonFiles.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+* Equip [Random Background](equipBGcolor.json) :  This JSON file contains the schema for equipping a random backgroundd from habitica
+
+The API call it responds to is : ``` https://habitica.com/api/v3/user/unlock?path=background.{color} ```
```
Comment (ieahleen) [inline json_files/jsonFiles.md:10]:
```suggestion
* Equip [Random Background](equipBGcolor.json) :  This JSON file contains the schema for equipping a random backgroundd from habitica

The API call it responds to is : `https://habitica.com/api/v3/user/unlock?path=background.{color}`
```

File: json_files/jsonFiles.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+* [Buy New Equpiment](buyNewEquip.json) : This JSON file contains the schema for buying a new quipment from the market in habitica. 
+
+The API call it responds to is: ```https://habitica.com/api/v3/user/buy-gear/{itemToPurchase.key}```
```
Comment (ieahleen) [inline json_files/jsonFiles.md:14]:
```suggestion
* [Buy New Equpiment](buyNewEquip.json) : This JSON file contains the schema for buying a new quipment from the market in habitica. 

The API call it responds to is: `https://habitica.com/api/v3/user/buy-gear/{itemToPurchase.key}`
```

File: json_files/jsonFiles.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+* Equip [Random Costume](costumeRandom.json) :  This JSON file contains the schema for equipping a random costume from habitica
+
+The API call it responds to is : ``` https://habitica.com/api/v3/user/equip/costume/{costume_name} ```
```
Comment (ieahleen) [inline json_files/jsonFiles.md:18]:
```suggestion
* Equip [Random Costume](costumeRandom.json) :  This JSON file contains the schema for equipping a random piece of equipment ascostume from habitica

The API call it responds to is : `https://habitica.com/api/v3/user/equip/costume/{costume_name}` 
```

File: json_files/jsonFiles.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+The API call it responds to is: ```https://habitica.com/api/v3/user/buy-gear/{itemToPurchase.key}```
+
+* Equip [Random Costume](costumeRandom.json) :  This JSON file contains the schema for equipping a random costume from habitica
+
+The API call it responds to is : ``` https://habitica.com/api/v3/user/equip/costume/{costume_name} ``` 
+
+[Random Battle Gear](battlegearRndm.json) : This JSON file contains the schema for equipping a random battle gear from habitica
+
+The API call it responds to is : ``` https://habitica.com/api/v3/user/equip/equipped/{item} ```
```
Comment (ieahleen) [inline json_files/jsonFiles.md:22]:
```suggestion
[Random Battle Gear](battlegearRndm.json) : This JSON file contains the schema for equipping a randompiece of equipment as battle gear from habitica

The API call it responds to is : `https://habitica.com/api/v3/user/equip/equipped/{item}`
```

**Review by ieahleen — CHANGES_REQUESTED** [review]
Thank you, I suggested a few changes to the `.md` file

**Comment by aswinikalyan30** [comment]
Updated them :)

**Review by ieahleen — CHANGES_REQUESTED** [review]
I think it's missing the user data API call and the Habitica content API call, if you please can add those too

Also this doesn't close #49, it's not related at all to #49, please change what issue you are referring to

**Comment by aswinikalyan30** [comment]
So which issue is this related to?

**Comment by aswinikalyan30** [comment]
#49 is the one for json schema

**Comment by ieahleen** [comment]
json schema of the tool, you are doing #36

**Comment by aswinikalyan30** [comment]
Updated with relevant changes

File: json_files/jsonFiles.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
+## Description of JSON files
+### This folder contains the json schema for the responses of the API calls made by habitican-randomiser
+
+* Equip [Random Pet](equipPet.json) : This JSON file contains the schema for equpping a random pet from habitica. 
+
+The API call it responds to is: `https://habitica.com/api/v3/user/equip/pet/{pet_name}`
```
Comment (ieahleen) [inline json_files/jsonFiles.md:6]:
```suggestion
* Equip [Random Pet](equipPet.json) is the response to `https://habitica.com/api/v3/user/equip/pet/{pet_name}`

This is the API endpoint to equip a new pet
```

File: json_files/jsonFiles.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+* Equip [Random Background](equipBGcolor.json) :  This JSON file contains the schema for equipping a random backgroundd from habitica
+
+The API call it responds to is : ` https://habitica.com/api/v3/user/unlock?path=background.{color} `
```
Comment (ieahleen) [inline json_files/jsonFiles.md:10]:
```suggestion
* Equip [Random Background](equipBGcolor.json) is the response to `https://habitica.com/api/v3/user/unlock?path=background.{color}`

This is the call to equip a background.
```

File: json_files/jsonFiles.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+* [Buy New Equpiment](buyNewEquip.json) : This JSON file contains the schema for buying a new quipment from the market in habitica. 
+
+The API call it responds to is: `https://habitica.com/api/v3/user/buy-gear/{itemToPurchase.key}`
```
Comment (ieahleen) [inline json_files/jsonFiles.md:14]:
```suggestion
* [Buy New Equpiment](buyNewEquip.json) is the response to `https://habitica.com/api/v3/user/buy-gear/{itemToPurchase.key}`

This is the API endpoint to buy a piece of gear.
```

File: json_files/jsonFiles.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+* Equip [Random Costume](costumeRandom.json) :  This JSON file contains the schema for equipping a random piece of equipment as costume from habitica
+
+The API call it responds to is : ` https://habitica.com/api/v3/user/equip/costume/{costume_name} `
```
Comment (ieahleen) [inline json_files/jsonFiles.md:18]:
```suggestion
* Equip [Random Costume](costumeRandom.json) is the response to `https://habitica.com/api/v3/user/equip/costume/{costume_name}`

It's the API endpoint to equip a piece of gear as costume
```

File: json_files/jsonFiles.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+* [Random Battle Gear](battlegearRndm.json) : This JSON file contains the schema for equipping a random piece of equipment as battle gear from habitica
+
+The API call it responds to is : ` https://habitica.com/api/v3/user/equip/equipped/{item} `
```
Comment (ieahleen) [inline json_files/jsonFiles.md:22]:
```suggestion
* [Random Battle Gear](battlegearRndm.json) is the response to `https://habitica.com/api/v3/user/equip/equipped/{item}`

Same as above but for battle gear
```

File: json_files/jsonFiles.md [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+* [User](user.json): This JSON file contains the details for a user in habitica
+
+The API call it responds to is : ` https://habitica.com/api/v3/user `
```
Comment (ieahleen) [inline json_files/jsonFiles.md:26]:
```suggestion
* [User](user.json) is the response to `https://habitica.com/api/v3/user`

This is the API call to get all the user data
```

File: json_files/jsonFiles.md [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+
+* [Content](content.json) : This JSON file contains the details of all the content used by a user in habitica
+
+The API call it responds to is : ` https://habitica.com/api/v3/content?language=en `
```
Comment (ieahleen) [inline json_files/jsonFiles.md:30]:
```suggestion
* [Content](content.json) is the response to `https://habitica.com/api/v3/content?language=en`

It's all available Habitica content.
```

**Comment by aswinikalyan30** [comment]
Done!

**Pull request merged**